Barotrauma/BarotraumaClient/ClientSource/GUI/GUITextBox.cs

```diff
@@ -756,7 +779,27 @@ public void ReceiveSpecialInput(Keys key)
                     HandleSelection();
                     break;
                 case Keys.Delete when !Readonly:
-                    if (selectedCharacters > 0)
+                    if (PlayerInput.IsCtrlDown())
+                    {
+                        var removeTo = FindStopSymbolRight(Text, CaretIndex);
+                        var len = removeTo - CaretIndex;
+                        if (len <= 0)
+                        {
+                            if (selectedCharacters > 0)
+                            {
+                                RemoveSelectedText();
+                            }
+
+                            break;
+                        }
+
+                        var newText = Text.Remove(CaretIndex, len);
+                        SetText(newText, false);
```

Comment:
Should be `SetText(newText, store: true)`, or undo/redo memento won't register this.